controllers/servingruntime_controller.go

```diff
@@ -369,23 +420,45 @@ func (r *ServingRuntimeReconciler) getRuntimesSupportingPredictor(ctx context.Co
 		return nil, err
 	}
 
+	// list all cluster serving runtimes
+	csrs := &kserveapi.ClusterServingRuntimeList{}
+	if r.HasNamespaceAccess {
+		if err := r.Client.List(ctx, csrs); err != nil {
+			return nil, err
+		}
+	}
+
 	restProxyEnabled := r.ConfigProvider.GetConfig().RESTProxy.Enabled
-	srnns := make([]types.NamespacedName, 0, len(runtimes.Items))
+	srNameMap := make(map[string]bool)
+	for i := range csrs.Items {
+		crt := &csrs.Items[i]
+		if crt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(&crt.Spec, p, restProxyEnabled, crt.Name) {
+			srNameMap[crt.GetName()] = true
+		}
+	}
 	for i := range runtimes.Items {
 		rt := &runtimes.Items[i]
-		if rt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(rt, p, restProxyEnabled) {
-			srnns = append(srnns, types.NamespacedName{
-				Name:      rt.GetName(),
-				Namespace: p.Namespace,
-			})
+		if rt.Spec.IsMultiModelRuntime() && runtimeSupportsPredictor(&rt.Spec, p, restProxyEnabled, rt.Name) {
+			srNameMap[rt.GetName()] = true
 		}
 	}
+	srnns := make([]types.NamespacedName, 0, len(srNameMap))
```

Comment:
We could consider changing this function to return the map rather than building a slice here, since the calling function just builds another slice.